homework/index.js

```diff
@@ -30,15 +33,189 @@
+  const container = createAndAppend('div', root, { id: 'container' });
+
+  function getSelectedData(value, data) {
+    let selected;
+    data.forEach(item => {
+      if (item.name === value) {
+        selected = item;
       }
     });
+    return selected;
+  }
+
+  function generateSelections(repos) {
+    const header = createAndAppend('header', root, {});
+    const selectionElem = createAndAppend('select', header, {
+      id: 'repositories',
+    });
+    let id = 0;
+
+    repos.forEach(repo => {
+      createAndAppend('option', selectionElem, {
+        id: id++,
+        value: repo,
+        text: repo,
+      });
+    });
+  }
+
+  function generateInfoSection(selected, data) {
+    const div = createAndAppend('div', container, { id: 'infoDiv' });
+    const selectedData = getSelectedData(selected, data);
+    createAndAppend('a', div, {
+      id: 'repoName',
+      text: `Repository: ${selectedData.name}`,
+      href: selectedData.html_url,
+      target: '_blank',
+    });
+    createAndAppend('p', div, {
+      id: 'desc',
+      text: `Description: ${selectedData.description}`,
+    });
+    createAndAppend('p', div, {
+      id: 'forks',
+      text: `Forks: ${selectedData.forks}`,
+    });
+    createAndAppend('p', div, {
+      id: 'updated_at',
+      text: `Updated: ${selectedData.updated_at}`,
+    });
+  }
+
+  function generateContributorsSection(selected) {
+    const contributorsDiv = createAndAppend('div', container, { id: 'contsDiv' });
+    createAndAppend('p', contributorsDiv, { id: 'cont-header', text: 'Contributors' });
+    const contributorsLists = createAndAppend('ul', contributorsDiv, { id: 'conts-list' });
+
+    fetchJSON(`https://api.github.com/repos/HackYourFuture/${selected}/contributors`).then(
+      data => {
+        data.forEach(item => {
+          const contributorListItem = createAndAppend('li', contributorsLists, {
+            id: `${item.login}ListItem`,
+            class: 'cont-li',
+          });
+
+          createAndAppend('img', contributorListItem, {
+            id: `${item.login}Img`,
+            class: 'cont-avatar',
+            src: item.avatar_url,
+          });
+
+          const contributorDiv = createAndAppend('div', contributorListItem, {
+            id: `${item.login}Div`,
+          });
+
+          createAndAppend('a', contributorDiv, {
+            id: `${item.login}Name`,
+            class: 'cont-name',
+            text: item.login,
+            href: item.html_url,
+            target: '_blank',
+          });
+          createAndAppend('div', contributorDiv, {
+            id: `${item.login}Badge`,
+            class: 'cont-badge',
+            text: item.contributions,
+          });
+        });
+      },
+      err => {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      },
+    );
+  }
+
+  function updateContributorSection(selected) {
+    const contributorsDiv = document.getElementById('contsDiv');
```

Comment:
This is interesting. If you compare both `generateContributorsSection` and `updateContributorSection` methods, you will see that except `contributorsDiv.innerHTML = '';` this statement, all other logic is exactly same.

If such is the case, I would suggest that you keep your code DRY(don't repeat yourself). You can inside `updateContributorSection`, after clearing out the innerHTML, call the `generateContributorsSection` function. This way, you are saving about 40 lines of code and that's a lot. 🙂